src/DssDirectDepositAaveDai.sol

```diff
@@ -244,268 +157,59 @@ contract DssDirectDepositAaveDai {
+        pool.deposit(wat, amt, msg.sender, 0);
 
-        if (vat.live() == 0) {
-            // MCD caged
-            require(EndLike(chainlog.getAddress("MCD_END")).debt() == 0, "DssDirectDepositAaveDai/end-debt-already-set");
-            require(culled == 0, "DssDirectDepositAaveDai/module-has-to-be-unculled-first");
-            _unwind(
-                type(uint256).max,
-                availableLiquidity,
-                Mode.MCD_CAGED
-            );
-        } else if (live == 0) {
-            // This module caged
-            _unwind(
-                type(uint256).max,
-                availableLiquidity,
-                culled == 1
-                ? Mode.MODULE_CULLED
-                : Mode.NORMAL
-            );
-        } else {
-            // Normal path
-            uint256 supplyAmount = _add(
-                                    availableLiquidity,
-                                    _add(
-                                        stableDebt.totalSupply(),
-                                        variableDebt.totalSupply()
-                                    )
-                                );
-            uint256 targetSupply = bar > 0 ? calculateTargetSupply(bar) : 0;
+    }
 
-            if (targetSupply > supplyAmount) {
-                _wind(targetSupply - supplyAmount);
-            } else if (targetSupply < supplyAmount) {
-                _unwind(
-                    supplyAmount - targetSupply,
-                    availableLiquidity,
-                    Mode.NORMAL
-                );
-            }
-        }
+    // Withdraws Dai from Aave in exchange for adai
+    // Aave: https://docs.aave.com/developers/v/2.0/the-core-protocol/lendingpool#withdraw
+    function withdraw(address wat, uint256 amt) external auth {
+        // We need to pull adai tokens in this address before calling withdraw
+        require(adai.transferFrom(msg.sender, address(this), amt), "DssDirectDepositAaveDai/withdraw-transfer-failed");
+        // Then we can withdraw and send the Dai to the msg.sender
+        pool.withdraw(wat, amt, msg.sender);
     }
 
-    // --- Collect Interest ---
-    function reap() external {
-        require(vat.live() == 1, "DssDirectDepositAaveDai/no-reap-during-shutdown");
-        require(live == 1, "DssDirectDepositAaveDai/no-reap-during-cage");
-        uint256 adaiBalance = adai.balanceOf(address(this));
-        (, uint256 daiDebt) = vat.urns(ilk, address(this));
-        if (adaiBalance > daiDebt) {
-            uint256 fees = adaiBalance - daiDebt;
-            uint256 availableLiquidity = dai.balanceOf(address(adai));
-            if (fees > availableLiquidity) {
-                fees = availableLiquidity;
-            }
-            pool.withdraw(address(dai), fees, address(this));
-            daiJoin.join(address(chainlog.getAddress("MCD_VOW")), fees);
-        }
+    function getStrategy(address wat) public view returns (address strategy) {
```

Comment:
I guess the asset `wat` will be always `dai`, right?

Reply:
good call, should be fixed in: https://github.com/makerdao/dss-direct-deposit/pull/31/commits/f44d186183e4c5a6e195263420e73a42151c545b